infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -157,6 +161,8 @@
     private Set<String> responsiveNodesPeerView = Collections.emptySet();
     // Connectivity of any responsive nodes not responding. Updated by FailureDetector.
     private Set<String> unresponsiveNodesPeerView = Collections.emptySet();
+    // Cluster View compiled by aggregating latest peer views from heartbeat responses.
+    private final ConcurrentHashMap<String, PeerView> clusterView = new ConcurrentHashMap<>();
```

Comment:
This doesn't seem to be thread safe. Although accessing single entries is thread-safe, making a copy isn't. getConnectivityView can be called from another thread while clusterView is being updated.